portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -140,9 +140,23 @@ private void buildCommentData() {
   }
 
   // Returns the JSON representation of the given Linked List.
-  private String convertToJson(ArrayList<Object[]> data) {
+  public String convertToJson(ArrayList<Object[]> data) {
     Gson gson = new Gson();
     String json = gson.toJson(data);
     return json;
   }
+
+  // BELOW IS ARE HELPER METHODS FOR THE TESTS THAT ARE RUN IN THE PortfolioTest.java FILE
+  
+  //initializes the data ArrayList.
+  public ArrayList<Object[]> initData() {
+    // Initialize List/s
+    data = new ArrayList<Object[]>();
+    buildIntroData();
+    buildCommentData();
+
+    // Return the internal data list.
+    return data;
+  }
+
```

Comment:
I would refrain from adding testing helpers in your actual code. Instead I would call doGet() in your test so the intro and comment data are built naturally. You want your tests to reflect the behavior in production and constructing test-only paths can potentially cause you to miss errors.